tests/Integration/Schema/Types/UnionTest.php

```diff
@@ -190,18 +190,16 @@ public static function fetchResults(): EloquentCollection
     }
 
     /** @return array<int, array<string>> */
-    public function withAndWithoutCustomTypeResolver(): array
+    public static function withAndWithoutCustomTypeResolver(): iterable
     {
-        return [
-            // This uses the default type resolver
-            $this->schemaAndQuery(false),
-            // This scenario requires a custom resolver, since the types User and Post do not match
-            $this->schemaAndQuery(true),
-        ];
+        // This uses the default type resolver
+        yield self::schemaAndQuery(false);
+        // This scenario requires a custom resolver, since the types User and Post do not match
+        yield self::schemaAndQuery(true);
```

Comment:
For better output on failing tests, datasets can be "named"

```suggestion
        yield "default type resolver" => self::schemaAndQuery(false);
        // This scenario requires a custom resolver, since the types User and Post do not match
        yield "custom type resolver" => self::schemaAndQuery(true);
```

On failing test, instead of 
```
There were 5 errors:

1) Tests\Integration\Schema\Types\UnionTest::testResolveUnionTypes with data set #1
```

You will get 
```
There were 5 errors:

1) Tests\Integration\Schema\Types\UnionTest::testResolveUnionTypes with data set "custom type resolver"
```